blockchain/chain_sync/src/network_context.rs

```diff
@@ -93,6 +114,38 @@ impl SyncNetworkContext {
         Ok(fts.remove(0))
     }
 
+    /// Requests that some content with a particular Cid get fetched over Bitswap if it doesn't
+    /// exist in the BlockStore.
+    pub async fn bitswap_get<TMessage: DeserializeOwned>(
+        &self,
+        content: Cid,
+    ) -> Result<TMessage, String> {
+        // TODO: check to see if we have it in our database
+        let (tx, rx) = oneshot_channel();
+        self.network_send
+            .send(NetworkMessage::BitswapRequest {
+                cid: content.clone(),
+                response_channel: tx,
+            })
+            .await;
+        let res = future::timeout(Duration::from_secs(RPC_TIMEOUT), rx).await;
+        match res {
+            Ok(Ok(())) => {
+                match self.db.get(&content) {
+                    Ok(Some(b)) => Ok(b),
+                    Ok(None) => Err(format!("Bitswap response successful for: {:?}, but can't find it in the database", content)),
+                    Err(e) => Err(format!("Bitswap response successful for: {:?}, but can't retreive it from the database: {}", content, e.to_string())),
+                }
+            }
+            Err(e) => {
+               Err(format!("Bitswap get for {:?} timed out: {}", content, e.to_string()))
+            }
+            Ok(Err(e)) => {
+                Err(format!("Bitswap get for {:?} failed: {}", content, e.to_string()))
+            }
+        }
```

Comment:
why does this have to be all matches when it's just returning errors or the result?

Reply:
Just wanted to return more concise err messages.

Reply:
This is more confusing imo, at least for reading, I don't see why you couldn't at least return the timeout error before this match